htdocs/install/mysql/tables/llx_asset_accountancy_codes_economic-asset.sql

```diff
@@ -0,0 +1,39 @@
+-- ========================================================================
+-- Copyright (C) 2022      OpenDSI              <support@open-dsi.fr>
+--
+-- This program is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU General Public License as published by
+-- the Free Software Foundation, either version 3 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU General Public License for more details.
+--
+-- You should have received a copy of the GNU General Public License
+-- along with this program.  If not, see https://www.gnu.org/licenses/.
+-- ========================================================================
+--
+-- Table to store the configuration of the accounting accounts of a fixed asset for economic status
+--
+-- Data example:
+-- INSERT INTO `llx_asset_accountancy_codes_economic` (`rowid`, `fk_asset`, `fk_asset_model`, `asset`, `depreciation_asset`, `depreciation_expense`, `value_asset_sold`, `receivable_on_assignment`, `proceeds_from_sales`, `vat_collected`, `vat_deductible`, `tms`, `fk_user_modif`) VALUES
+-- (2, 1, NULL, '2183', '2818', '68112', '675', '465', '775', '44571', '44562', '2022-01-18 14:20:20', 1);
+
+CREATE TABLE llx_asset_accountancy_codes_economic(
+    rowid						integer			AUTO_INCREMENT PRIMARY KEY NOT NULL,
+    fk_asset					integer,
+    fk_asset_model				integer,
```

Comment:
My understanding is that each "asset" as a type into "asset_model".
So you have a link with fk_asset_model into table fk_asset.

The table llx_asset_accountancy_codes_economic has record linked to an asset.
So we already have an inherited link to the asset model through the asset. The model is a property of the asset.
So we should not have fk_asset_model here.
Should be the same for other tables.

Reply:
No, not quite.

An asset can be created with or without a model.

If you create a template, you will be able to set up accounting accounts for the economic & fiscal status in this template.
When you create an asset, you choose this model and the accounting accounts of the model are included in this asset because not everyone knows all the accounting accounts to correctly account for an asset.
In the table llx_asset_accountancy_codes_economic, you will have a row for the model and another for the asset created.

If you create an asset without a template, you can set all the accounting accounts on or off.
In the table llx_asset_accountancy_codes_economic, you will have a row for the asset created.

In summary, models & assets both use the tables llx_asset_accountancy_codes_economic & llx_asset_accountancy_codes_fiscal to store their information. If it's a model, fk_asset_model is populated. If it is an asset, only fk_asset is filled.

![image](https://user-images.githubusercontent.com/2341395/169680393-a8d5f115-caf7-43dc-984d-bf28db0ebc3a.png)

I correct some code for model here: #20984

Reply:
so let's start with your summary
"If it's a model, fk_asset_model is populated. If it is an asset, only fk_asset is filled."
So it means the table is used to store accountancy code for 2 differents object. Why not if property are very similar.

So into file llx_asset_accountancy_code_economic.sql, you have this comment:

`-- Table to store the configuration of the accounting accounts of a fixed asset for economic status`

If my understanding is good, can we replace it with

```
-- Table to store the configuration of the accounting accounts of a fixed asset for economic status (fk_asset will be filled and fk_asset_model will be null)
-- or to store the configuration of the accounting accounts for templates of asset (fk_asset_model will be filled and fk_asset will be null)
```

Can you confirm ?